### PR TITLE
Add ERC 20 example contract to highlight HIP 218

### LINKED ISSUE #32: Add ERC 20 example contract to highlight HIP 218
### Problem

[HIP 218](https://hips.hedera.com/hip/hip-218) brought support to allow erc-20-calls-directly-to-token-accounts.
However, example references don't exist in code base for easily reference by devs to deploy and tests

### Solution

Add solidity contracts similar to [ERC20Contract](https://github.com/hashgraph/hedera-json-rpc-relay/pull/184/files#diff-669f58b84685c5d798d1799f55efe3eac1528d38c36f1230207a09152232745d) to highlight [HIP 218](https://hips.hedera.com/hip/hip-218#erc-20-calls-directly-to-token-accounts)

See https://docs.hedera.com/guides/core-concepts/smart-contracts/supported-erc-token-standards#examples, this should be in this repo not primarily in services repo

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
Signed-off-by: nikolay <n.atanasow94@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #32

Depends on https://github.com/hashgraph/hedera-smart-contracts/pull/74

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/hts-precompile/ERC20Contract.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+contract ERC20Contract {
```
Comment (Nana-EC) [inline contracts/hts-precompile/ERC20Contract.sol:7]: Looks good.
Please wait on https://github.com/hashgraph/hedera-smart-contracts/pull/57 to go in and then add test coverage for at least the `name`, `symbol`, `decimals` and `totalSupply` methods.

Reminder the goal would be the initial coverage we added in the relay would not be needed and this repo would be the source of basic and simple contracts with their tests using hardhat and local-node.
Nothing complex, just easy to view and run from a pure web3 (less Hedera) point of view

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looks good, just needs the tests coverage setup using hardhat and local node

**Comment by shemnon** [comment]
I would move this to the examples directory, based on the tag line of the PR.

**Comment by Nana-EC** [comment]
> Reference
Yeah #74 might have to go in first to provide the folder structure

File: contracts/hts-precompile/ERC20Contract.sol (hunk 1 above)
Comment (Nana-EC) [inline contracts/hts-precompile/ERC20Contract.sol:7]: To Danno's point this should be under examples
Place it under `contracts/hts-precompile/examples` in an `erc20` dir

**Review by Nana-EC — COMMENTED** [review]
Looks good, just need to move ERC20Contract.sol into it's own dir under `contrcts/hts-precompile/examples`

**Comment by shemnon** [comment]
Files should be in a location with "example" as a directory in the path.

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**